pandas/_libs/tslibs/nattype.pyi

```diff
@@ -15,7 +18,12 @@ nat_strings: set[str]
 
 def is_null_datetimelike(val: object, inat_is_null: bool = ...) -> bool: ...
 
-class NaTType(datetime):
+_NaTComparisonTypes = Union[datetime, timedelta, Period, np.datetime64, np.timedelta64]
+
+class _NatComparison:
+    def __call__(self, other: _NaTComparisonTypes) -> bool: ...
```

Comment:
This is needed because when you compare to `np.datetime64` and `np.timedelta64`, their comparison operators are declared this way.  `mypy` complains when just having typing stubs (in the Microsoft stubs I'm working on), and this resolves the issue there and here in pandas.